### ISSUE TITLE
UI Fabric override the color provided by theme color

### ISSUE THREAD

**Issue by PooLP** [estruyf/react-handlebars-search#16]
Hi,

If use a ui-fabric color class on my markup (ex:ms-fontColor-themePrimary), the preloaded ui fabric override the color provided by theme color (see screenshot)

![ui-fabric](https://user-images.githubusercontent.com/582733/37640170-ab7c06d4-2c14-11e8-8f3a-38d87a1c5e48.jpg)

I have two idea :

The simple way is add option in panel for desactivate import for ui fabric core. I can add this approach quickly.

The another way, is add a handlebar variable system to use the ui fabric theme color

ex : 
.my class {
 color: {{styles.themePrimary}}
}

But, i don't tested this way.

**Comment by estruyf** [comment]
Another way might be to override all the theme classes in our SCSS file. 

```
.ms-fabric--v6-0-0 .ms-fontColor-themePrimary {
  color: $ms-color-themePrimary;
}
```

It is indeed a bit stupid, but having the UI Fabric Core styles available for the template is a key thing.

**Comment by PooLP** [comment]
If you want, I would test my second way this week because we will be able to use the colors on other css property (ex: animation, gradient ...), if I can not do it, I'll take your way. I'm go back after test.

**Comment by estruyf** [comment]
Of course you could test it, but using a helper means that the person who is going to create the template would not be able to use the class names from the UI Fabric documentation.

When we override them, they can just use the class names without any helper, which is a bit easier, in the end, don't you think?

**Comment by PooLP** [comment]
I agree. I can quickly taking your way and later test/add my way ? it may be the best solution :) ... my way requires testing and verification not to weigh down the package, optimisation ... both can be used in final package or it's bad idea ?

**Comment by PooLP** [comment]
Hi,
I have tested your idea, but didn't work, the css is correctly generated, but the colors it's the default colors, not a theme colors.

My sass file to import in your SASS file : 

`
/* themed tokens */
$colors: (
    themeDarker: "[theme:themeDarker, default: #004578]",
    themeDark: "[theme:themeDark, default: #005a9e]",
    themeDarkAlt: "[theme:themeDarkAlt, default: #106ebe]",
    themePrimary: "[theme:themePrimary, default: #0078d7]",
    themeSecondary: "[theme:themeSecondary, default: #2b88d8]",
    themeTertiary: "[theme:themeTertiary, default: #71afe5]",
    themeLight: "[theme:themeLight, default: #c7e0f4]",
    themeLighter: "[theme:themeLighter, default: #deecf9]",
    themeLighterAlt: "[theme:themeLighterAlt, default: #eff6fc]",
    black: "[theme:black, default: #000000]",
    neutralDark: "[theme:neutralDark, default: #212121]",
    neutralPrimary: "[theme:neutralPrimary, default: #333333]",
    neutralPrimaryAlt: "[theme:neutralPrimaryAlt, default: #3c3c3c]",
    neutralSecondary: "[theme:neutralSecondary, default: #666666]",
    neutralSecondaryAlt: "[theme:neutralSecondaryAlt, default: #767676]",
    neutralTertiary: "[theme:neutralTertiary, default: #a6a6a6]",
    neutralTertiaryAlt: "[theme:neutralTertiaryAlt, default: #c8c8c8]",
    neutralQuaternary: "[theme:neutralQuaternary, default: #d0d0d0]",
    neutralQuaternaryAlt: "[theme:neutralQuaternaryAlt, default: #dadada]",
    neutralLight: "[theme:neutralLight, default: #eaeaea]",
    neutralLighter: "[theme:neutralLighter, default: #f4f4f4]",
    neutralLighterAlt: "[theme:neutralLighterAlt, default: #f8f8f8]",
    white: "[theme:white, default: #ffffff]",
    blackTranslucent40: "[theme:blackTranslucent40, default: rgba(0,0,0,.4)]",
    whiteTranslucent40: "[theme:whiteTranslucent40, default: rgba(255,255,255,.4)]",
    yellow: "[theme:yellow, default: #ffb900]",
    yellowLight: "[theme:yellowLight, default: #fff100]",
    orange: "[theme:orange, default: #d83b01]",
    orangeLight: "[theme:orangeLight, default: #ff8c00]",
    redDark: "[theme:redDark, default: #a80000]",
    red: "[theme:red, default: #e81123]",
    magentaDark: "[theme:magentaDark, default: #5c005c]",
    magenta: "[theme:magenta, default: #b4009e]",
    magentaLight: "[theme:magentaLight, default: #e3008c]",
    purpleDark: "[theme:purpleDark, default: #32145a]",
    purple: "[theme:purple, default: #5c2d91]",
    purpleLight: "[theme:purpleLight, default: #b4a0ff]",
    blueDark: "[theme:blueDark, default: #002050]",
    blueMid: "[theme:blueMid, default: #00188f]",
    blue: "[theme:blue, default: #0078d7]",
    blueLight: "[theme:blueLight, default: #00bcf2]",
    tealDark: "[theme:tealDark, default: #004b50]",
    teal: "[theme:teal, default: #008272]",
    tealLight: "[theme:tealLight, default: #00b294]",
    greenDark: "[theme:greenDark, default: #004b1c]",
    green: "[theme:green, default: #107c10]",
    greenLight: "[theme:greenLight, default: #bad80a]",
    error: "[theme:error, default: #a80000]",
    errorText: "[theme:errorText, default: #333333]",
    errorBackground: "[theme:errorBackground, default: #fde7e9]",
    success: "[theme:success, default: #107c10]",
    successText: "[theme:successText, default: #333333]",
    successBackground: "[theme:successBackground, default: #dff6dd]",
    alert: "[theme:alert, default: #d83b01]",
    alertText: "[theme:alertText, default: #333333]",
    alertBackground: "[theme:alertBackground, default: #deecf9]",
    warning: "[theme:warning, default: #767676]",
    warningText: "[theme:warningText, default: #333333]",
    warningBackground: "[theme:warningBackground, default: #fff4ce]",
    severeWarning: "[theme:severeWarning, default: #d83b01]",
    severeWarningText: "[theme:severeWarningText, default: #333333]",
    severeWarningBackground: "[theme:severeWarningBackground, default: #fed9cc]",
    info: "[theme:info, default: #767676]",
    infoText: "[theme:infoText, default: #333333]",
    infoBackground: "[theme:infoBackground, default: #f4f4f4]",
    orangeLighter: "[theme:orangeLighter, default: #ea4300]"
);

:global {
    // Fix for theme color
    .ms-Fabric--v6-0-0 {
        @each $name, $color-value in $colors {
            .ms-fontColor-#{$name} {
                color: $color-value;
                &--hover {
                    color: $color-value;
                }
            }
            .ms-bgColor-#{$name} {
                background-color: $color-value;
                &--hover {
                    background-color: $color-value;
                }
            }
            .ms-borderColor-#{$name} {
                border-color: $color-value;
                &--hover {
                    border-color: $color-value;
                }
            }
        }
    }
}
`
I do not know why I have this behavior.

**Comment by PooLP** [comment]
I'm sorry, it's working if package is generated, but didn't work with `gulp serve`

**Comment by estruyf** [comment]
Closing the issue as your code got merged #17
